geo/src/algorithm/coords_iter.rs

```diff
@@ -111,6 +155,11 @@ impl<'a, T: CoordinateType + 'a> CoordsIter<'a, T> for MultiLineString<T> {
     fn coords_iter(&'a self) -> Self::Iter {
         MapCoordsIter(self.0.iter(), marker::PhantomData).flatten()
     }
+
+    /// Return the number of coordinates in the `MultiLineString`.
+    fn coords_count(&'a self) -> usize {
+        self.0.iter().map(|line_string| line_string.num_coords()).sum()
```

Comment:
It seems unfortunate to have both a `coords_count` method and a `num_coords` method - especially since num_coords doesn't exist for most geometries. I can easily imagine getting them confused and annoyed as I reach for it and it's only sometimes there.

What would you think of either:

1. renaming `coords_count` to `num_coords`
-or-
2. leaving them as `coords_count` but deprecating `num_coords`

Reply:
Does `_count` implies some `O(n)` operation? 🤔

Reply:
oh i meant to change this to `line_string.coords_count()` within the closure on this line. i didn't want to remove `num_coords` because that would be a geo-types breaking change and i didn't want to deal with that. the standard library uses `count` and `count_*`, but not `num_`, so that's why i changed it (along with [your earlier suggestion](https://github.com/georust/geo/pull/563#pullrequestreview-547626299)). so i think i'm slightly in favor of your second option.

> Does `_count` imply some `O(n)` operation? 🤔

i don't think so! the default `Iterator#count` implementation is `O(n)`, but many iterators overwrite it to be `O(1)`, [like `std::vec::IntoIter`](https://github.com/rust-lang/rust/blob/6340607acaab10eed3cf11ef7ad3564db58ec981/library/alloc/src/vec.rs#L3179-L3182) (e.g. `vec.into_iter().count()`)

Reply:
ok! That sounds good to me. This all LGTM. Thanks for you patience with my review. 😬